### PR TITLE
Created model card for XLM model

### PR DESCRIPTION
**Pull request by AshAnand34**
# What does this PR do?

This pull request updates the XLM model documentation to improve clarity, add usage examples, and enhance the overall presentation. The changes include a new layout for badges, detailed usage examples with code snippets, and reorganized content for better readability as outlined in #36979.

### Documentation Enhancements:

* Added a floating badge section for PyTorch and TensorFlow at the top of the page to visually highlight supported frameworks.
* Included detailed usage examples for masked language modeling using both the `Pipeline` and `AutoModel` classes, complete with Python code snippets.

### Content Reorganization:

* Rewrote and reorganized the "Overview" and "Model Details" sections to improve clarity and readability. Key features and training objectives are now presented in a concise bullet-point format.
* Moved usage tips to a dedicated "Usage Tips" section, emphasizing the importance of selecting the correct checkpoint and using the `lang` parameter for multilingual tasks.

### Additional Features:

* Added a "TIP" section to guide users to explore more examples of XLM applications via the sidebar.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/xlm.md [hunk 1]
```diff
@@ -16,44 +16,91 @@ rendered properly in your Markdown viewer.
...
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
+<div style="float: right;">
+    <div class="flex flex-wrap space-x-1">
+        <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
+        <img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
+    </div>
 </div>
 
-## Overview
+[XLM](https://arxiv.org/abs/1901.07291) is a transformer model pretrained using one of three objectives: causal language modeling (CLM), masked language modeling (MLM), or translation language modeling (TLM). What makes XLM unique is its ability to handle multiple languages through cross-lingual pretraining, achieving state-of-the-art results on cross-lingual classification and machine translation tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md]:
```suggestion
[XLM](https://huggingface.co/papers/1901.07291) demonstrates cross-lingual pretraining with two approaches, unsupervised training on a single language and supervised training on more than one language with a cross-lingual language model objective. The XLM model supports the causal language modeling objective, masked language modeling, and translation language modeling (an extension of the [BERT](./bert)) masked language modeling objective to multiple language inputs).

is a transformer model pretrained using one of three objectives: causal language modeling (CLM), masked language modeling (MLM), or translation language modeling (TLM). What makes XLM unique is its ability to handle multiple languages through cross-lingual pretraining, achieving state-of-the-art results on cross-lingual classification and machine translation tasks.
```

File: docs/source/en/model_doc/xlm.md [hunk 2]
```diff
@@ -16,44 +16,91 @@ rendered properly in your Markdown viewer.
...
 
-The XLM model was proposed in [Cross-lingual Language Model Pretraining](https://arxiv.org/abs/1901.07291) by
-Guillaume Lample, Alexis Conneau. It's a transformer pretrained using one of the following objectives:
+You can find all the original XLM checkpoints under the [XLM](https://huggingface.co/models?search=xlm) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md]:
```suggestion
You can find all the original XLM checkpoints under the [Facebook AI community](https://huggingface.co/FacebookAI?search_models=xlm-mlm) organization.
```

**Review by stevhliu — COMMENTED** [review]
Thanks! See my comments on your other PRs and apply them to this one as well :)

File: docs/source/en/model_doc/xlm.md (hunk 2 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm.md]: Fixed.

File: docs/source/en/model_doc/xlm.md (hunk 1 above)
Comment (AshAnand34) [reply docs/source/en/model_doc/xlm.md]: Just fixed per the PR comments on xlm-roberta-xl and xlm-roberta

File: docs/source/en/model_doc/xlm.md [hunk 3]
```diff
@@ -14,55 +14,76 @@ rendered properly in your Markdown viewer.
...
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
-</div>
+[XLM](https://huggingface.co/papers/1901.07291) demonstrates cross-lingual pretraining with two approaches, unsupervised training on a single language and supervised training on more than one language with a cross-lingual language model objective. The XLM model supports the causal language modeling objective, masked language modeling, and translation language modeling (an extension of the [BERT](./bert)) masked language modeling objective to multiple language inputs).
+
+You can find all the original XLM checkpoints under the [Facebook AI community](https://huggingface.co/FacebookAI?search_models=xlm-mlm) organization.
+
+> [!TIP]
+> Click on the XLM models in the right sidebar for more examples of how to apply XLM to different cross-lingual tasks like classification, translation, and question answering.
+
+The example below demonstrates how to predict the `[MASK]` token with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md]:
```suggestion
The example below demonstrates how to predict the `<mask>` token with [`Pipeline`], [`AutoModel`] and from the command line.
```

File: docs/source/en/model_doc/xlm.md [hunk 4]
```diff
@@ -14,55 +14,76 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```python
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md]: Replace with:

```py
import torch  
from transformers import pipeline  

pipeline = pipeline(  
    task="fill-mask",  
    model="facebook/xlm-roberta-xl",  
    torch_dtype=torch.float16,  
    device=0  
)  
pipeline("Bonjour, je suis un modèle <mask>.")  
```

File: docs/source/en/model_doc/xlm.md [hunk 5]
```diff
@@ -14,55 +14,76 @@ rendered properly in your Markdown viewer.
...
+```
 
-- a causal language modeling (CLM) objective (next token prediction),
-- a masked language modeling (MLM) objective (BERT-like), or
-- a Translation Language Modeling (TLM) object (extension of BERT's MLM to multiple language inputs)
+</hfoption>
+<hfoption id="AutoModel">
 
-The abstract from the paper is the following:
+```python
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md:54]: Replace with:

```py
import torch  
from transformers import AutoModelForMaskedLM, AutoTokenizer  

tokenizer = AutoTokenizer.from_pretrained(  
    "FacebookAI/xlm-mlm-en-2048",  
)  
model = AutoModelForMaskedLM.from_pretrained(  
    "FacebookAI/xlm-mlm-en-2048",  
    torch_dtype=torch.float16,  
    device_map="auto",  
)  
inputs = tokenizer("Hello, I'm a <mask> model.", return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    predictions = outputs.logits.argmax(dim=-1)

predicted_token = tokenizer.decode(predictions[0][inputs["input_ids"][0] == tokenizer.mask_token_id])
print(f"Predicted token: {predicted_token}")
```

File: docs/source/en/model_doc/xlm.md [hunk 6]
```diff
@@ -14,55 +14,76 @@ rendered properly in your Markdown viewer.
...
+# Decode prediction
+predicted_token = tokenizer.decode(predictions[0][inputs["input_ids"][0] == tokenizer.mask_token_id])
+print(f"Predicted token: {predicted_token}")
+```
 
-    * Causal language modeling (CLM) which is the traditional autoregressive training (so this model could be in the previous section as well). One of the languages is selected for each training sample, and the model input is a sentence of 256 tokens, that may span over several documents in one of those languages.
-    * Masked language modeling (MLM) which is like RoBERTa. One of the languages is selected for each training sample, and the model input is a sentence of 256 tokens, that may span over several documents in one of those languages, with dynamic masking of the tokens.
-    * A combination of MLM and translation language modeling (TLM). This consists of concatenating a sentence in two different languages, with random masking. To predict one of the masked tokens, the model can use both, the surrounding context in language 1 and the context given by language 2.
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md]: Add `transformers-cli` option

```bash
echo -e "Plants create <mask> through a process known as photosynthesis." | transformers-cli run --task fill-mask --model FacebookAI/xlm-mlm-en-2048 --device 0
```

File: docs/source/en/model_doc/xlm.md [hunk 7]
```diff
@@ -14,55 +14,76 @@ rendered properly in your Markdown viewer.
...
 
-## Resources
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/xlm.md]: Remove all the `## Notes`

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution!
